xlators/performance/readdir-ahead/src/readdir-ahead.c

```diff
@@ -697,6 +699,16 @@ rda_fill_fd(call_frame_t *frame, xlator_t *this, fd_t *fd)
 
     UNLOCK(&ctx->lock);
 
+    /* if the file is sharded then server will return the below
+     * xattr as part of dict that is used to update the ia_size
+     * in d_stat
+     */
+    ret = dict_set_uint64(ctx->xattrs, GF_XATTR_SHARD_FILE_SIZE, 8 * 4);
```

Comment:
What about other xattrs that other xlators may require ?

It doesn't seem a good idea to explicitly use knowledge from other xlators inside readdir-ahead. This kills modularity.

Reply:
Is there any other way, I can fetch the shard_flie_size in this part of the code ?. Since the file size is stored as part of the xattr for shared file, If we want to enable readdir-ahead/paraller-readdir, then we need to fetch the xattr in this part of the code.

Reply:
I think you can set the xattr at the time of calling opendir by shard.

Reply:
Sure. I will send the xdata in shard_open() and check

Reply:
Update the patch with this suggestion